pkg/devfile/parser/util/utils_test.go

```diff
@@ -89,8 +89,8 @@ func TestDownloadInMemoryClient(t *testing.T) {
 			wantErr: fmt.Sprintf(downloadErr, "https://"+RawGitHubHost+"/devfile/library/main/devfile.yaml"),
 		},
 		{
-			name:   "Case 6: Input url is valid with a mock client",
-			client: MockDevfileUtilsClient{MockGitURL: util.MockGitUrl{Host: server.URL}, DownloadOptions: util.MockDownloadOptions{MockFile: "OK"}},
+			name:   "Case 6: Input url is valid with a mock client, dont use mock data during invocation",
+			client: MockDevfileUtilsClient{},
 			url:    server.URL,
```

Comment:
Do you mean a Github URL? I prefer if we stayed with the test server since we are comparing bytes here. And the string will be unnecessarily long. I think thats one reason whoever wrote this test first did the test server approach to test short files.

Reply:
Since this is a non-token test, the mock implementation uses the regular implementatino to download the URL. And since its not a Github URL, the regular implementation does a simple GET.

Reply:
since you add support for empty client, should add new unit test for your change, not editing the existing  tests. the empty client case you added, needs a gitURL?

Reply:
I'll add a Github URL